libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -3632,26 +3632,17 @@ MAV_RESULT GCS_MAVLINK::handle_command_int_do_set_home(const mavlink_command_int
     if (!is_zero(packet.param1)) {
         return MAV_RESULT_FAILED;
     }
-    // check frame type is supported
-    if (packet.frame != MAV_FRAME_GLOBAL &&
-        packet.frame != MAV_FRAME_GLOBAL_INT &&
-        packet.frame != MAV_FRAME_GLOBAL_RELATIVE_ALT &&
-        packet.frame != MAV_FRAME_GLOBAL_RELATIVE_ALT_INT) {
+    Location::ALT_FRAME frame;
+    if (!mavlink_coordinate_frame_to_location_alt_frame(packet.frame, frame)) {
+        // unknown coordinate frame
         return MAV_RESULT_UNSUPPORTED;
     }
-    Location new_home_loc {};
-    new_home_loc.lat = packet.x;
-    new_home_loc.lng = packet.y;
-    new_home_loc.alt = packet.z * 100;
-    // handle relative altitude
-    if (packet.frame == MAV_FRAME_GLOBAL_RELATIVE_ALT ||
-        packet.frame == MAV_FRAME_GLOBAL_RELATIVE_ALT_INT) {
-        if (!AP::ahrs().home_is_set()) {
-            // cannot use relative altitude if home is not set
-            return MAV_RESULT_FAILED;
```

Comment:
we're losing this check

Reply:
I've added a change to always convert home into an absolute altitude in the AHRS `set_home` call.  This may fail if we have no home set.

Reply:
I don't think that is sufficient. It's a cyclic and badly defined operation to update home based on home, and adding weird cases where it does/doesn't work does not make sense. I really think relative altitudes should always be rejected here.

Reply:
.... you know it was like that when I got here, right?  The check I've added catches the original check but for all callers rather than just this one.

Also - I'm not disagreeing with you in terms of offset-from-home.  Relative offsets from other places might be OK, 'though.  OTOH, I don't know why we obviously allowed in in the first place...